From 7e74623513fa170fb568f149a703cfc489965ebf Mon Sep 17 00:00:00 2001
From: James Cameron <quozl@laptop.org>
Date: Thu, 9 Feb 2017 10:18:54 +1100
Subject: [PATCH] Defend against Gtk3 cursor new_from_name bug

On Ubuntu 16.10 and later, Gdk.Cursor raises a TypeError instead of
returning None as documented.

Fixes

    File "/usr/share/sugar/activities/Paint.activity/Area.py", line 1754, in
    set_tool_cursor
        cursor = Gdk.Cursor.new_from_name(display, name)
    TypeError: constructor returned NULL

Documentation reference:

https://lazka.github.io/pgi-docs/Gdk-3.0/classes/Cursor.html#Gdk.Cursor.new_from_name
---
 Area.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Area.py b/Area.py
index db227f8..77ecc05 100644
--- a/Area.py
+++ b/Area.py
@@ -1772,7 +1772,7 @@ def set_tool_cursor(self):
 
                 cursor = Gdk.Cursor.new_from_pixbuf(display, pixbuf,
                                                     hotspot_x, hotspot_y)
-        except GObject.GError:
+        except GObject.GError, TypeError:
             cursor = None
         if self.get_window() is not None:
             self.get_window().set_cursor(cursor)
